qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -629,11 +629,6 @@ def tensor(self, other):
         """
         if not isinstance(other, PauliList):
             other = PauliList(other)
-        if len(other) not in [1, len(self)]:
```

Comment:
I'm not very familiar with the quantum info part, but this error does not apply anymore since you changed the `_tensor` method below, right?

Reply:
This is mistake when we implement PauliList. This error check is unnecessary because the tensor is well-defined such as  [A, B] ^ [C, D, E] = A^C + A^D + ...